packages/codec/lib/compilations/utils.ts

```diff
@@ -145,9 +145,13 @@ export function shimContracts(
     //ast needs to be coerced because schema doesn't quite match our types here...
 
     //if files or sources was passed, trust that to determine the source index
-    if (files || inputSources) {
+    //(assuming we have a sourcePath!)
```

Comment:
I think it's worth explaining the correlation between sourcePath and Solidity version as an extended comment.

Reply:
I agree with Amal's comment of explaining the correlation between the two but up to your determination.

Reply:
OK, I'll add something quickly before merge.